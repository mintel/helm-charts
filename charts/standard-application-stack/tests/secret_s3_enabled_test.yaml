suite: Test S3 External Secret
templates:
  - secrets.yaml
release:
  namespace: test-namespace
tests:
  - it: Check S3 remoteRef key is present
    set:
      global.clusterEnv: qa
      global.name: test-app
      s3.enabled: true
      # Disable default application secret
      externalSecret.enabled: false
    asserts:
      - isKind:
          of: ExternalSecret
      - equal:
          path: metadata.namespace
          value: test-namespace
      - hasDocuments:
          count: 1
      - equal:
          path: spec.data
          value:
            - remoteRef:
                key: test-namespace/test-app/s3
                property: BUCKET_NAME
              secretKey: BUCKET_NAME
  - it: Check S3 secretPath override
    set:
      global.clusterEnv: qa
      global.name: test-app
      s3.enabled: true
      s3.secretPathOverride: overridePath
      # Disable default application secret
      externalSecret.enabled: false
    asserts:
      - isKind:
          of: ExternalSecret
      - equal:
          path: metadata.namespace
          value: test-namespace
      - hasDocuments:
          count: 1
      - equal:
          path: spec.data
          value:
            - remoteRef:
                key: overridePath
                property: BUCKET_NAME
              secretKey: BUCKET_NAME
  - it: Check S3 refresh interval override
    set:
      global.clusterEnv: qa
      global.name: test-app
      s3.enabled: true
      s3.secretRefreshIntervalOverride: 10m5s
      # Disable default application secret
      externalSecret.enabled: false
    asserts:
      - isKind:
          of: ExternalSecret
      - equal:
          path: metadata.namespace
          value: test-namespace
      - hasDocuments:
          count: 1
      - equal:
          path: spec.refreshInterval
          value: 10m5s
  - it: Check S3 Secret Store Override
    set:
      global.clusterEnv: qa
      global.name: test-app
      s3.enabled: true
      s3.secretStoreRefOverride: some-other-secret-store
      # Disable default application secret
      externalSecret.enabled: false
    asserts:
      - isKind:
          of: ExternalSecret
      - equal:
          path: metadata.namespace
          value: test-namespace
      - hasDocuments:
          count: 1
      - equal:
          path: spec.secretStoreRef.name
          value: some-other-secret-store
  - it: Check S3 Secret is present for local environment
    set:
      global.clusterEnv: local
      global.name: test-app
      s3.enabled: true
      # Disable default application secret
      externalSecret.enabled: false
    asserts:
      - isKind:
          of: Secret
      - equal:
          path: metadata.namespace
          value: test-namespace
      - hasDocuments:
          count: 1
      - equal:
          path: data.BUCKET_NAME
          value: dGVzdC1hcHAtczM=
