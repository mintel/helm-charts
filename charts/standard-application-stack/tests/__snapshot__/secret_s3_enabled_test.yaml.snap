Check S3 Secret Store Override:
  1: |
    apiVersion: external-secrets.io/v1beta1
    kind: ExternalSecret
    metadata:
      annotations:
        argocd.argoproj.io/sync-options: SkipDryRunOnMissingResource=true
      labels:
        app.kubernetes.io/component: app
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: test-app
        app.mintel.com/env: qa
        app.mintel.com/region: ${CLUSTER_REGION}
        name: test-app
      name: test-app-s3
      namespace: test-namespace
    spec:
      data:
      - remoteRef:
          key: test-namespace/test-app/s3
          property: BUCKET_NAME
        secretKey: BUCKET_NAME
      refreshInterval: 5m0s
      secretStoreRef:
        kind: SecretStore
        name: some-other-secret-store
      target:
        creationPolicy: Owner
Check S3 Secret is present for local environment:
  1: |
    apiVersion: v1
    data:
      BUCKET_NAME: dGVzdC1hcHAtczM=
    kind: Secret
    metadata:
      annotations: null
      labels:
        app.kubernetes.io/component: app
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: test-app
        app.mintel.com/env: local
        app.mintel.com/region: local
        name: test-app
      name: test-app-s3
      namespace: test-namespace
    type: Opaque
Check S3 refresh interval override:
  1: |
    apiVersion: external-secrets.io/v1beta1
    kind: ExternalSecret
    metadata:
      annotations:
        argocd.argoproj.io/sync-options: SkipDryRunOnMissingResource=true
      labels:
        app.kubernetes.io/component: app
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: test-app
        app.mintel.com/env: qa
        app.mintel.com/region: ${CLUSTER_REGION}
        name: test-app
      name: test-app-s3
      namespace: test-namespace
    spec:
      data:
      - remoteRef:
          key: test-namespace/test-app/s3
          property: BUCKET_NAME
        secretKey: BUCKET_NAME
      refreshInterval: 10m5s
      secretStoreRef:
        kind: SecretStore
        name: aws-secrets-manager-default
      target:
        creationPolicy: Owner
Check S3 remoteRef key is present:
  1: |
    apiVersion: external-secrets.io/v1beta1
    kind: ExternalSecret
    metadata:
      annotations:
        argocd.argoproj.io/sync-options: SkipDryRunOnMissingResource=true
      labels:
        app.kubernetes.io/component: app
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: test-app
        app.mintel.com/env: qa
        app.mintel.com/region: ${CLUSTER_REGION}
        name: test-app
      name: test-app-s3
      namespace: test-namespace
    spec:
      data:
      - remoteRef:
          key: test-namespace/test-app/s3
          property: BUCKET_NAME
        secretKey: BUCKET_NAME
      refreshInterval: 5m0s
      secretStoreRef:
        kind: SecretStore
        name: aws-secrets-manager-default
      target:
        creationPolicy: Owner
Check S3 secretPath override:
  1: |
    apiVersion: external-secrets.io/v1beta1
    kind: ExternalSecret
    metadata:
      annotations:
        argocd.argoproj.io/sync-options: SkipDryRunOnMissingResource=true
      labels:
        app.kubernetes.io/component: app
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: test-app
        app.mintel.com/env: qa
        app.mintel.com/region: ${CLUSTER_REGION}
        name: test-app
      name: test-app-s3
      namespace: test-namespace
    spec:
      data:
      - remoteRef:
          key: overridePath
          property: BUCKET_NAME
        secretKey: BUCKET_NAME
      refreshInterval: 5m0s
      secretStoreRef:
        kind: SecretStore
        name: aws-secrets-manager-default
      target:
        creationPolicy: Owner
